Python/ceval.c

```diff
@@ -203,13 +204,25 @@ _PyEval_FiniThreads(void)
     }
 }
 
+static inline void
+exit_thread_if_finalizing(PyThreadState *tstate)
+{
+    /* _Py_Finalizing is protected by the GIL */
+    if (_Py_IsFinalizing() && !_Py_CURRENTLY_FINALIZING(tstate)) {
+        drop_gil(tstate);
+        PyThread_exit_thread();
+        Py_UNREACHABLE();
```

Comment:
Remark: I am not sure why Py_UNREACHABLE() is needed here. Isn't PyThread_exit_thread() marked with Py_NO_RETURN?

Reply:
@ncoghlan Is `Py_UNREACHABLE()` necessary here ?

Reply:
I suspect Victor is right, and this code simply predates `PyThread_exit_thread()` being marked as `Py_NO_RETURN`.

Having both is harmless though, and serves as a clear indicator to the reader, even if the compiler doesn't need it.

Reply:
I created https://github.com/python/cpython/pull/13068 to fix this issue